src/api/alchemy-websocket-provider.ts

```diff
@@ -120,6 +130,24 @@ export class AlchemyWebSocketProvider
     this.cancelBackfill = noop;
   }
 
+  /**
+   * Overrides the `BaseProvider.getNetwork` method as implemented by ethers.js.
+   *
+   * This override allows the SDK to set the provider's network to values not
+   * yet supported by ethers.js.
+   *
+   * @internal
+   * @override
+   */
+  static getNetwork(network: Networkish): NetworkFromEthers {
```

Comment:
Possibly make this a private util class to avoid duplicate code?

Reply:
This needs to be public so that the super constructor code ends up finding this override ([context](https://github.com/ethers-io/ethers.js/blob/master/packages/providers/src.ts/json-rpc-provider.ts#L432): this is where it traverses the object to find the first `static` call to this method)